app/src/marketplace/modules/communityProduct/services.js

```diff
@@ -0,0 +1,75 @@
+// @flow
+
+import { deploy } from '$mp/utils/smartContract'
+import getConfig from '$shared/web3/config'
+
+import type { SmartContractDeployTransaction } from '$shared/flowtype/web3-types'
+import type {
+    StreamId,
+    Stream,
+    NewStream,
+} from '$shared/flowtype/stream-types'
+import type { Permission } from '$userpages/flowtype/permission-types'
+import type { ApiResult } from '$shared/flowtype/common-types'
+
+import { post } from '$shared/utils/api'
+import { formatApiUrl } from '$shared/utils/url'
+import { postStream } from '$userpages/modules/userPageStreams/services'
+
+export const getStreamrEngineAddresses = (): Array<string> => {
+    const addressesString = process.env.STREAMR_ENGINE_NODE_ADDRESSES || ''
+    const addresses = addressesString.split(',')
+    return addresses
+}
+
+export const addPermission = (id: StreamId, permission: Permission): ApiResult<Array<Permission>> =>
+    post(formatApiUrl('streams', id, 'permissions'), permission)
+
+export const addKey = (id: StreamId, key: Object): ApiResult<Object> =>
+    post(formatApiUrl('streams', id, 'keys'), key)
+
+export const createJoinPartStream = async (productName: string): Promise<Stream | null> => {
+    const newStream: NewStream = {
+        name: `JoinPart stream for ${productName}`,
+        description: 'Automatically created JoinPart stream for community product contract',
+    }
+    const stream = await postStream(newStream)
+    if (stream == null) {
+        console.error('Could not create JoinPart stream')
+        return null
+    }
+
+    // Add public read permission
+    try {
+        await addPermission(stream.id, {
+            anonymous: true,
+            operation: 'read',
+            user: null,
+        })
+    } catch (e) {
+        console.error('Could not add public read permission for JoinPart stream', e)
+    }
+
+    // Add write permissions for all Streamr Engine nodes
+    try {
+        const addWriteKeyPromises = getStreamrEngineAddresses().map((address) => (
+            addKey(stream.id, {
+                name: address,
+                permission: 'write',
+            })
+        ))
+        await Promise.all(addWriteKeyPromises)
+    } catch (e) {
+        console.error('Could not add write keys to JoinPart stream', e)
+    }
+
+    return stream
```

Comment:
Third step added: Remove own share permission to make the stream show up as read-only in stream lists.

Reply:
For production and staging this is wise. I'm not sure if prohibiting delete is wise for local dev environment because there will be tons of streams created when playing around in the UI. What do you think? Adding isProduction check here feels a bit weird :/

Reply:
I think better to have to same behaviour in all environments. Locally we can just wipe the database once in a while if gets unmanageable.

Reply:
Left out share permission removing for now since backend will return an error when we try to delete the only 'share' permission.